server/src/main/java/org/opensearch/action/bulk/TransportBulkAction.java

```diff
@@ -603,6 +610,12 @@ protected void doRun() {
 
             final AtomicInteger counter = new AtomicInteger(requestsByShard.size());
             String nodeId = clusterService.localNode().getId();
+
+            final AtomicLong[] docStatusCounter = new AtomicLong[5];
```

Comment:
Do we need to use `AtomicLongArray`?

Reply:
I think we may need to make the array thread safe as well to ensure the AtomicLong provided by array remain synchronized on access across threads